[Fan Jue] iP

FYimu opened the pull request:
# Preview to Tesseract

> "We look beyond the Earth to greater worlds the Tesseract will unveil." - The Other from [The Avengers](https://www.imdb.com/title/tt0848228/characters/nm1089991)

Tesseract helps to track what you need to do and what you have done. It's
* Text-based
* Easy to learn
* ~~FAST~~*SUPER FAST* to use

As a starter, you just need to
1. Download it from [here]()
2. Double-click it
3. Use it
And it will manage your tasks for you 😜

It is powerful and **FREE**!

Available commands for now: `list`, `delete`, `mark`, `unmark`, `filter`, `find`, `event`, `todo`, `deadline`, `exit`

## FEATURES
- [x] Managing tasks
- [ ] Reminders (coming soon)
- [ ] Ordering tasks according to priority (coming soon)

## Change memory address
If you are familiar with Java codes, you can change where Tesseract stores your task history in the Tesseract class here by replacing `src/main/Data/Schedule.txt` with the file path you want 😉 
```ruby
/**
 * Main driving class of Tesseract.
 * @author Fan Jue
 * @version 0.1.0
 * @since 0.1.0
 */
public class Tesseract {
    static final String SCHEDULE_PATH = "src/main/Data/Schedule.txt";
    ...
```

Commit:
Add Gradle support

FYimu committed:
Tesseract -version "wk2.lvl1"

FYimu committed:
I forgot to commit level 2 and 3 QAQ

FYimu committed:
Tesseract -version("w2.l2")

FYimu committed:
Tesseract -version("w2.l3")

FYimu committed:
Tesseract -version("w2.l4.2")

FYimu committed:
Modification for previous levels

FYimu committed:
Increment #5: Handle errors

FYimu committed:
Increment #6: Delete

FYimu committed:
Add function <Save>

FYimu committed:
Add Date specification and command <filter>

FYimu committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add Date specification and command <filter>

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Tesseract.java

FYimu committed:
Meet Stretch Goal

FYimu committed:
Organise into three packages: tesseract.main, tesseract.task, tesseract.command

FYimu committed:
Add Gradle Support

FYimu committed:
Add Gradle

FYimu committed:
Create JAR file for Tesseract

FYimu committed:
Add JavaDoc for all non-private classes/methods

FYimu committed:
Adjust according to doing standards

FYimu committed:
Add function <find>

FYimu committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

* branch-A-JavaDoc:
  Add JavaDoc for all non-private classes/methods

# Conflicts:
#	src/main/java/Task.java
#	src/main/java/TaskList.java

FYimu committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Adjust according to doing standards

# Conflicts:
#	src/main/java/Command.java
#	src/main/java/TessUi.java

FYimu committed:
Set up Gradle

FYimu committed:
Update package

FYimu committed:
Set up CheckStyle.

FYimu committed:
Modify according to CheckStyle instruction

punnyhuimin commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,162 @@
+     * @param str The input string to be tested.
+     * @return True if the input string represents an integer, false otherwise.
+     */
+    public static boolean isInteger(String str) {
```
Good effort in the naming conventions for methods that return booleans 👍

punnyhuimin commented on src/main/java/command/CreateTaskCommand.java:
```diff
@@ -0,0 +1,69 @@
+ * @since 0.1.0
+ */
+public class CreateTaskCommand extends Command {
+    /** The type of task to be created */
```
Good effort in providing an explanation for class members 😄

punnyhuimin commented on src/main/java/command/ExitCommand.java:
```diff
@@ -0,0 +1,50 @@
+ * @since 0.1.0
+ */
+public class ExitCommand extends Command {
+    /** The specific command keyword */
```
I think you forgot to delete this comment?

FYimu replied:
Yes haha good catch!

punnyhuimin commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+     * @param tNew A task to be added into the TaskList.
+     */
+    public void addTask(Task tNew) {
+        this.numOfTasks += 1;
```
Is there a need to use "this" if there are no overlapping keywords?

punnyhuimin commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+
+    /**
+     * Return a string representation of the list of all tasks to be printed out.
+     * @return A string representation of the TaskList.
```
Maybe you can put @\return on a newline instead? You can get Intellij to do [auto-formatting](https://stackoverflow.com/questions/17879475/how-enable-auto-format-code-for-intellij-idea) for you.
<img width="377" alt="Screenshot 2022-02-03 at 8 14 02 PM" src="https://user-images.githubusercontent.com/61443625/152340762-fbaa532b-b84e-4e2f-88be-73711512e1ec.png">

FYimu replied:
Thanks for your useful trick!

punnyhuimin commented on src/main/java/main/TessUi.java:
```diff
@@ -0,0 +1,124 @@
+    /**
+     * Display the welcome message to the user.
+     */
+    void sayHi() {
```
Should you declare this as private or public?

punnyhuimin commented on src/main/java/main/TessUi.java:
```diff
@@ -0,0 +1,124 @@
+                + BREAKER);
+    }
+
+    void display(String msg) {
```
Should you declare this as private or public?

punnyhuimin replied:
Is it necessary to put JavaDocs here?

punnyhuimin commented on src/main/java/main/TessUi.java:
```diff
@@ -0,0 +1,124 @@
+
+    /**
+     * Display the list of tasks to the user.
+     */
```
Are you missing an explanation for the method's input?

punnyhuimin commented on src/main/java/main/TessUi.java:
```diff
@@ -0,0 +1,124 @@
+    /**
+     * Display the deleted task to the user.
+     */
+    public void deleteTaskRes(String msg, int numOfTask) {
```
Do you need to include the input explanations for the method?

punnyhuimin commented on src/main/java/main/TessUi.java:
```diff
@@ -0,0 +1,124 @@
+
+    /**
+     * Display the added task to the user.
+     */
```
Is there a missing explanation for the input parameter? Could it be the same for the rest of the methods in this file?

punnyhuimin commented on src/main/java/main/Tesseract.java:
```diff
@@ -0,0 +1,59 @@
+
+    /**
+     * Main driver of the AI.
+     */
```
Is there a missing explanation for the method's inputs?

punnyhuimin commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    /**
+<<<<<<< HEAD
```
Is this necessary?

punnyhuimin reviewed (commented):
LGTM! Minor issues with the JavaDocs, but great effort overall with the variable names and method naming 💯

DALIN-Prog commented on src/main/java/main/Tesseract.java:
```diff
@@ -0,0 +1,59 @@
+            try {
+                String fullCmd = ui.readCommand();
+                Command.process(fullCmd, taskList.size()); // throw any possible error
+                Command cmd = Command.generate(fullCmd); // command is guaranteed to be valid
```
If I'm not mistaken, perhaps you should put your comments above the code i.e
// throw any possible error
Command.process(fullCmd, taskList.size()); 
// command is guaranteed to be valid
Command cmd = Command.generate(fullCmd);

FYimu replied:
True true that looks cleaner 👍

DALIN-Prog commented on src/main/java/main/TessUi.java:
```diff
@@ -0,0 +1,124 @@
+ */
+public class TessUi {
+    // deals with interactions with the user
+    protected static final String INDENT_ONE = "    ";
+    protected static final String INDENT_TWO = "        ";
```
For indentation, I would recommend appending "\t" for INDENT_ONE and "\t\t" for INDENT_TWO

FYimu replied:
Wow time to pick up a useful trick~

DALIN-Prog commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+     *
+     * @return A number representing the number of all current tasks.
+     */
+    public int size() {
```
Perhaps, it would be better if it's named getSize

DALIN-Prog commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+     * @param index The position of the queried task.
+     * @return Return the queried task.
+     */
+    public Task get(int index) {
```
Likewise, getTask here

DALIN-Prog commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+     * @param date Date on which the task happens.
+     * @return The list of tasks filtered.
+     */
+    public TaskList filterByDate(Date date) {
+        List<Task> tasks = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.isOn(date)) {
+                tasks.add(task);
+            }
+        }
+        return new TaskList(tasks, tasks.size());
+    }
+
+    /**
+     * Return a new TaskList of tasks filtered by the given keyword.
+     *
+     * @param keyword The keyword to filter all match task description.
+     * @return A TaskList instance containing all tasks containing the keyword.
+     */
+    public TaskList filterByKeyword(String keyword) {
+        List<Task> tasks = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.isMatch(keyword)) {
+                tasks.add(task);
+            }
+        }
+        return new TaskList(tasks, tasks.size());
+    }
```
Nice!

DALIN-Prog commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,179 @@
+        this.taskList = new ArrayList<Task>();
+        for (String str : storageMemory) {
+            String[] tasks = str.split("@", 4); // format: D@1@do something
+            Task taskNew;
```
Do you think if newTask would sound better?

DALIN-Prog commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,162 @@
+package command;
+
+import main.Date;
+import main.Storage;
+import main.TaskList;
+import main.TessUi;
+import main.TesseractException;
```
Thumbs up for stating all classes imported

FYimu committed:
Add GUI for Tesseract

DALIN-Prog reviewed (commented):
Overall, it looks like you have adhered closely to the coding standard. Great!

FYimu committed:
Add (belated) branch-Level-10

FYimu committed:
Assertion: document assumptions

Assertions can help to check if assumptions hold at various points in the code.

Insert assertions in all classes where applicable.

Let's use more assertions to facilitate debugging process.

FYimu committed:
Code quality: update codes according to guidelines on code quality

Code quality is essential for enhancing code readability to audience and making it more convenient to debug.

Remove magic literals and add in static literals; reorganise nested loops.

Let's pay more attention to code quality especially for output String to be displayed in the dialog box.

FYimu committed:
Use Stream for filter by date and filter by keyword.

FYimu committed:
Merge pull request #1 from FYimu/branch-A-Assertions

Assertion: document assumptions

FYimu committed:
Use Stream for filter by date and filter by keyword.

FYimu committed:
Merge branch 'master' into branch-A-CodeQuality

FYimu committed:
Merge pull request #2 from FYimu/branch-A-CodeQuality

Code quality: update codes according to guidelines on code quality

FYimu committed:
Merge branch 'master' of https://github.com/FYimu/ip

* 'master' of https://github.com/FYimu/ip:
  Code quality: update codes according to guidelines on code quality

FYimu committed:
Set up custom workflow - gradle.yml

FYimu committed:
Add extension B-ViewSchedules and C-Archive

FYimu committed:
Debug for archive

FYimu committed:
Add better Gui.

FYimu committed:
Fix bugs.

FYimu committed:
Update user guide

FYimu committed:
Refactored all files for generation of jar file
